ArcGISRuntimeSDKQt_CppSamples/Analysis/FindServiceAreasForMultipleFacilities/FindServiceAreasForMultipleFacilities.cpp

```diff
@@ -0,0 +1,178 @@
+}
+
+FindServiceAreasForMultipleFacilities::~FindServiceAreasForMultipleFacilities() = default;
+
+void FindServiceAreasForMultipleFacilities::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FindServiceAreasForMultipleFacilities>("Esri.Samples", 1, 0, "FindServiceAreasForMultipleFacilitiesSample");
+}
+
+MapQuickView* FindServiceAreasForMultipleFacilities::mapView() const
+{
+  return m_mapView;
+}
+
+bool FindServiceAreasForMultipleFacilities::taskRunning() const
+{
+  return m_taskRunning;
+}
+
+// Set the view (created in QML)
+void FindServiceAreasForMultipleFacilities::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_serviceAreasOverlay);
+
+  connect(m_facilitiesFeatureLayer, &FeatureLayer::doneLoading, this, [this](Error loadError){
+    if (!loadError.isEmpty())
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+
+    if (m_facilitiesFeatureLayer->loadStatus() == LoadStatus::Loaded)
+    {
+      // zoom to the full extent of the feature layer
+      m_mapView->setViewpointGeometry(m_facilitiesFeatureLayer->fullExtent(), 100);
+    }
+  });
+
+  emit mapViewChanged();
+}
+
+void FindServiceAreasForMultipleFacilities::findServiceAreas()
+{
+  // start showing indicator
+  m_taskRunning = true;
+  emit taskRunningChanged();
+
+  m_serviceAreaTask = new ServiceAreaTask(serviceAreaTaskUrl, this);
+  m_serviceAreaTask->load();
+
+  // once service area task is done loading, create default parameters
+  connect(m_serviceAreaTask, &ServiceAreaTask::doneLoading, this, [this](Error loadError)
+  {
+    if (!loadError.isEmpty())
+    {
+      qDebug() << loadError.message() << loadError.additionalMessage();
+      return;
+    }
+    m_serviceAreaTask->createDefaultParameters();
+  });
+
+  // once default parameters created, set parameters and solve
+  connect(m_serviceAreaTask, &ServiceAreaTask::createDefaultParametersCompleted, this, [this](QUuid, ServiceAreaParameters serviceAreaParameters)
+  {
+    serviceAreaParameters.setPolygonDetail(ServiceAreaPolygonDetail::High);
+    serviceAreaParameters.setReturnPolygons(true);
+
+    QList<double> travelTimes = {1.0, 3.0};
+
+    // add service areas of 1 minute and 3 minutes travel time by car
+    serviceAreaParameters.setDefaultImpedanceCutoffs(travelTimes);
+
+    // create query parameters used to select all facilities from the feature table
+    QueryParameters queryParameters;
+    queryParameters.setWhereClause("1=1");
+
+    // add all facilities to the service area parameters
+    serviceAreaParameters.setFacilitiesWithFeatureTable(m_facilitiesTable, queryParameters);
+
+    TaskWatcher taskWatcher = m_serviceAreaTask->solveServiceArea(serviceAreaParameters);
```

Comment:
If you store the `TaskWatcher` as a member you can use that to check whether the task is in flight - rather than the `m_taskRunning` member

Reply:
Agreed on this point.